### PR TITLE
Define optional targets as components

### PR DESCRIPTION
**Pull request by tpadioleau**
- DDC targets can be requested using components in `find_package`
- variables `DDC_BUILD_*` are removed, they are replaced by cmake components variables `DDC_*_FOUND`

### CODE REVIEW

File: install_test/CMakeLists.txt [hunk 1]
```diff
@@ -5,13 +5,13 @@
...
-find_package(DDC REQUIRED)
+find_package(DDC REQUIRED COMPONENTS fft pdi splines)
 
 message("DDC options:")
 message("DDC_BUILD_DOUBLE_PRECISION=${DDC_BUILD_DOUBLE_PRECISION}")
-message("DDC_BUILD_KERNELS_FFT=${DDC_BUILD_KERNELS_FFT}")
-message("DDC_BUILD_KERNELS_SPLINES=${DDC_BUILD_KERNELS_SPLINES}")
-message("DDC_BUILD_PDI_WRAPPER=${DDC_BUILD_PDI_WRAPPER}")
+message("DDC_fft_FOUND=${DDC_fft_FOUND}")
+message("DDC_splines_FOUND=${DDC_splines_FOUND}")
```
Comment (rb214678) [inline install_test/CMakeLists.txt:13]: Maybe a small cosmetic improvement if you put this line below in order to keep the order "fft pdi splines" that you seem to use elsewhere.

Comment (tpadioleau) [reply install_test/CMakeLists.txt:13]: True I tried to follow this rule, i will modify

File: install_test/CMakeLists.txt [hunk 2]
```diff
@@ -5,13 +5,13 @@
...
+message("DDC_pdi_FOUND=${DDC_pdi_FOUND}")
```
Comment (tpadioleau) [inline install_test/CMakeLists.txt:14]:
```suggestion
message("DDC_fft_FOUND=${DDC_fft_FOUND}")
message("DDC_pdi_FOUND=${DDC_pdi_FOUND}")
message("DDC_splines_FOUND=${DDC_splines_FOUND}")
```

**Review by rb214678 — APPROVED** [review]
Looks good to me now.

**Comment by tpadioleau** [comment]
> Looks good to me now.

Thanks I will merge after #671

**Pull request merged**